src/App.tsx

```diff
@@ -1,7 +1,37 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+export class App extends React.PureComponent {
+  state = {
+    pressedKey: null,
+  };
+
+  componentDidMount(): void {
+    document.addEventListener('keyup', (event: KeyboardEvent) => (
+      this.handleKeyUp(event)
+    ));
+  }
+
+  componentWillUnmount(): void {
+    document.removeEventListener('keyup', (event: KeyboardEvent) => (
+      this.handleKeyUp(event)
+    ));
+  }
```

Comment:
You are passing a newly created arrow funcion to document.addEventListener and document.removeEventListener. This will not work, pass handleKeyUp directly